src/main/java/org/apache/mesos/offer/DefaultOfferRequirementProvider.java

```diff
@@ -29,8 +29,9 @@ public DefaultOfferRequirementProvider(TaskConfigRouter taskConfigRouter) {
     @Override
     public OfferRequirement getNewOfferRequirement(String taskType, TaskSpecification taskSpecification)
             throws InvalidRequirementException {
+
         Protos.CommandInfo updatedCommand = taskConfigRouter.getConfig(taskType)
-                .updateEnvironment(taskSpecification.getCommand());
+                .updateEnvironment(taskSpecification.getCommand().get());
```

Comment:
Oh! That's a big miss. I also want to understand that why is there a special update method for the `CommandInfo`. Why not simply set it into the new `Protos.TaskInfo.Builder`?